### PR TITLE
Fixed inconsistent_struct_constructor triggers in macro-generated code

### LINKED ISSUE #7069: inconsistent_struct_constructor triggers in macro-generated code
Lint name: inconsistent_struct_constructor

I tried this code: Not that easy to provide since it requires a proc-macro crate, but title should be clear.

I expected to see this happen: No lints about inconsistent struct construction within macro-generated code.

Instead, this happened:

one example (this happened to me when working on [Ruma](https://github.com/ruma/ruma)):

```
warning: inconsistent struct constructor
  --> ruma-client-api/src/r0/backup/add_backup_key_session.rs:9:1
   |
9  | / ruma_api! {
10 | |     metadata: {
11 | |         description: "Store several keys in the backup.",
12 | |         method: PUT,
...  |
46 | |     error: crate::Error
47 | | }
   | |_^
```

### Meta

- `cargo clippy -V`: clippy 0.1.52 (07e0e2e 2021-03-24)
- `rustc -Vv`:
  ```
  rustc 1.53.0-nightly (07e0e2ec2 2021-03-24)
  binary: rustc
  commit-hash: 07e0e2ec268c140e607e1ac7f49f145612d0f597
  commit-date: 2021-03-24
  host: x86_64-unknown-linux-gnu
  release: 1.53.0-nightly
  LLVM version: 12.0.0
  ```

### PR DESCRIPTION
**Pull request by yawara**
fixes #7069 

changelog: `inconsistent_struct_constructor`: Fix FP in macro expansion.

### CODE REVIEW

File: clippy_lints/src/inconsistent_struct_constructor.rs [hunk 1]
```diff
@@ -66,6 +66,7 @@ declare_lint_pass!(InconsistentStructConstructor => [INCONSISTENT_STRUCT_CONSTRU
 impl LateLintPass<'_> for InconsistentStructConstructor {
     fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx hir::Expr<'_>) {
         if_chain! {
+            if !expr.span.from_expansion();
```
Comment (Y-Nak) [inline clippy_lints/src/inconsistent_struct_constructor.rs:69]: You could use `clippy_utils::in_macro` for this purpose, though this wouldn't affect the result in the case.

**Review by Y-Nak — COMMENTED** [review]
Could you add a test, please?

**Comment by llogiq** [comment]
Yeah, while this change seems quite trivial, a test would certainly be a good idea.

**Comment by llogiq** [comment]
Thank you!

@bors r+

**Pull request merged**